From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Wed, 15 Nov 2017 23:07:12 +0100
Subject: [PATCH] {BUG in 4.15-rc1} fs/btrfs/ref-verify: us of uninitiliazed
 variable root

The code is incorrect and clang complains:

    fs/btrfs/ref-verify.c:1017:22: error: variable 'root' is
    uninitialized when used here [-Werror,-Wuninitialized]
                    ret = walk_up_tree(root, path, &level);
                                       ^~~~
    fs/btrfs/ref-verify.c:987:25: note: initialize the variable 'root'
    to silence this warning
            struct btrfs_root *root;
                                   ^
                                    = NULL
---
 fs/btrfs/ref-verify.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/fs/btrfs/ref-verify.c b/fs/btrfs/ref-verify.c
index 34878699d363..3c47af01fd64 100644
--- a/fs/btrfs/ref-verify.c
+++ b/fs/btrfs/ref-verify.c
@@ -984,7 +984,6 @@ void btrfs_free_ref_tree_range(struct btrfs_fs_info *fs_info, u64 start,
 int btrfs_build_ref_tree(struct btrfs_fs_info *fs_info)
 {
 	struct btrfs_path *path;
-	struct btrfs_root *root;
 	struct extent_buffer *eb;
 	u64 bytenr = 0, num_bytes = 0;
 	int ret, level;
@@ -1014,7 +1013,7 @@ int btrfs_build_ref_tree(struct btrfs_fs_info *fs_info)
 				     &bytenr, &num_bytes);
 		if (ret)
 			break;
-		ret = walk_up_tree(root, path, &level);
+		ret = walk_up_tree(fs_info->extent_root, path, &level);
 		if (ret < 0)
 			break;
 		if (ret > 0) {
-- 
